posts/2024-05-21_our_experience_a.../our_experience_as_new_admiral_developers.qmd

```diff
@@ -0,0 +1,85 @@
+
+The beginning of Github can be scary and may seem complicated due to the potential for mistakes which might impact the entire project. While going through the processes and learning how it exactly works, it’s most of the time a simple process, and while working on the right parallel branch, nothing serious can happen. This is the most important point of working with Github: work on the right parallel branch, never/rarely on the main branch directly! 
+
+This is also where lots of discussions take place. Indeed, this is the place where the issues/tasks are officially described and stored.  
+
+# How you upskilled 
+
+During this open-source experience, we have had the opportunity to dive into a couple of existing issues that have expanded our technical knowledge (various packages: devtools, pkgdown, etc. themselves used for the creation of packages) and sharpened our problem-solving abilities. We received guidance from {admiral} team’s expertise on how to address the issues and improve our implementation. They helped us improve the quality of our code and broaden our understanding of best practices in development with their insights and suggestions.  
+
+{admiral} is not only developing code. It is also the development of the corresponding documentation and vignettes, as well as maintaining the online user guideline. This is also a forum where colleagues and users share opinions, what they think, how they would resolve an issue or how it is handled in their own company. There’s also a lot of communication around individual topics. 
+
+While completing tasks or resolving bugs which were identified in existing functions, it requires digging into the corresponding functions and learning how exactly they work. By becoming an {admiral} developer, we also become an {admiral} user. 
+
+The tasks are numerous and can correspond to any programming level. 
+
+# What have been the challenges/learnings 
+
+In terms of challenges, setting up the environment was time consuming. Of course, the main environment is company dependent, but following the first setup, there can be challenges to make the {admiral} package work as expected.  
+
+Some packages are required to run single steps. As the renv package has been deprecated for the use of {admiral} earlier this year, a single step is now necessary: *devtools::install_dev_deps(dependencies = TRUE)*. You just need to know it. Once all necessary packages are well installed and launched, it can be very easy to start developing some code or resolve some issues. 
+
+Another challenge is to understand the automatic CI/CD (Continuous Integration / Continuous Delivery) checks and their dependencies with other packages such as {admiraldev}. The CI/CD checks are made not to break down the current code while being updated. Some of these tests are implemented within the {admiraldev} package and used in {admiral} while creating a Pull request. Those updates can thus have some impacts on the {admiral} package itself. 
```

Comment:
```suggestion
Another challenge is to understand the automatic CI/CD (Continuous Integration / Continuous Delivery) checks and their dependencies with other packages such as {admiraldev}. The CI/CD checks are used to ensure any code updates we make won't break anything else in the codebase. These checks are implemented within the {admiralci} package and used in {admiral} when you create a Pull request. Those updates can thus have some impacts on the {admiral} package itself. 
```
For consideration, as i think reads a little clearer

Reply:
Is the first {admiraldev} correct or should it also be replaced by {admiralci} ?
![image](https://github.com/pharmaverse/blog/assets/157114584/cbad74ec-a082-4d20-9dea-33805e3cb9ea)

I wrote that because at some point of time some tests (testthat) were updated in {admiraldev} and crashing the CI/CD checks in {admiral}  🤔

Reply:
thats ok. a big dependency of admiral is admiraldev for the utility functions, but the CI checks themselves are in admiralci